EIPS/eip-2935.md

```diff
@@ -158,37 +153,43 @@ A special synthetic address is generated by working backwards from the desired d
+- for fork activation at block `10000`, block `9999`'s hash will be written to slot `9999`. Additional history for the the `BLOCKHASH_OLD_WINDOW` (`9744..9998`) will continue to be served, but will not be backfilled into storage. History for the blocks preceding the `HISTORY_SERVE_WINDOW` (`1808..9998`) will not be served, and will not be backfilled into storage. At block `20000`, the BLOCKHASH opcode would continue serving history for the `BLOCKHASH_OLD_WINDOW` (`19744..19999`) and would return 0 for blocks before the `BLOCKHASH_OLD_WINDOW`. The blocks from `HISTORY_SERVE_WINDOW` would be served from state for blocks `11808..19999`. Blocks prior to the `HISTORY_SERVE_WINDOW` would return 0.
 
- * For the fork to be activated at genesis, no history is written to the genesis state, and at the start of block `1`, genesis hash will be written as a normal operation to slot `0`.
- * for activation at block `1`, only genesis hash will be written at slot `0` as there is no additional history that needs to be persisted.
- * for activation at block `32`, block `31`'s hash will be written to slot `31` and additonal history for `0..30`'s hashes will be persisted, so all in all `0..31`'s hashes.
- * for activation at block `10000`, block `1808-9999`'s hashes will be persisted in the slot and `BLOCKHASH` for `1807` or less would resolve to `0` as only `HISTORY_SERVE_WINDOW` are persisted.
+For verkle-fork (blocks where `block.timestamp >= VERKLE_FORK_TIMESTAMP`), we consider that the fork (`FORK_TIMESTAMP`) is already in effect for at least `HISTORY_SERVE_WINDOW` blocks (or from genesis if `verkle_fork_block < HISTORY_SERVE_WINDOW`). The following changes take effect:
+
+- `BLOCKHASH` MUST retrieve blockhashes from the contract storage
+- The gas cost of the `BLOCKHASH` opcode is updated as per specification
+- The storage access is provided in the verkle execution witness.
 
 ### [EIP-158](./eip-158.md) handling
 
 This address is currently exempt from [EIP-158](./eip-158.md) cleanup in Kaustinen Verkle Testnet but we plan to address this in the following way:
 
-* Deploy a contract à la [EIP-4788](./eip-4788.md) which just supports `get` method to resolve the BLOCKHASH as per the logic defined in `resolve_blockhash` (and use the generated address as the BLOCKHASH contract address).
-* While the clients are expected to directly read from state (or maintain and serve from memory) to resolve BLOCKHASH opcode, this contract's `get` could be invoked by transaction (via another contract or directly) leading to a normal contract execution (and gas consumption) as per the semantics of the contract call.
-
+- Deploy a contract à la [EIP-4788](./eip-4788.md) which just supports `get` method to resolve the BLOCKHASH as per the logic defined in `resolve_blockhash` (and use the generated address as the BLOCKHASH contract address).
```

Comment:
should have a `set` as well

Reply:
Is this for 2935, or for the verkle one?